process/process.go

```diff
@@ -639,10 +639,14 @@ func replaceProcSelfWithProcPid(path string, pid uint32, tid uint32) string {
 
 	pidMatch := regexp.MustCompile(`^/proc/self/(.*)`)
 	tidMatch := regexp.MustCompile(`^/proc/self/task/[0-9]+/(.*)`)
+	tidMatch2 := regexp.MustCompile(`^/proc/thread-self/(.*)`)
 
 	repl = fmt.Sprintf("/proc/self/task/%d/${1}", tid)
 	p = tidMatch.ReplaceAllString(path, repl)
 
+	repl = fmt.Sprintf("/proc/self/task/%d/${1}", tid)
```

Comment:
If I understand this properly, does it make sense to run lines 645 and 648 in sequence? Shouldn't we run either 645 or 648?

Reply:
It's going to be one or the other anyway because the regexp don't overlap. In other words, either the `tidMatch` regex hits or the `tidMatch2` regex hits, but both can never hit (i.e., the first regex matches paths starting with `/proc/self/task`, the second with `/proc/thread-self/`). Thus I kept it that way, instead of adding an extra `if-else` clause.

Reply:
Got it.